src/Features/Core/Portable/InitializeParameter/AbstractInitializeMemberFromParameterCodeRefactoringProviderMemberCreation.cs

```diff
@@ -122,18 +125,49 @@ internal abstract partial class AbstractInitializeMemberFromParameterCodeRefacto
             }
         }
 
-        private IFieldSymbol CreateField(
-            IParameterSymbol parameter, ImmutableArray<NamingRule> rules, ImmutableArray<string> parameterNameParts)
+        private async Task<IFieldSymbol> CreateFieldAsync(
+            Document document, SyntaxNode functionDeclaration, IParameterSymbol parameter, ImmutableArray<NamingRule> rules, ImmutableArray<string> parameterNameParts, CancellationToken cancellationToken)
         {
+            var options = await document.GetOptionsAsync(cancellationToken).ConfigureAwait(false);
+            var requireAccessibilityModifiers = options.GetOption(CodeStyleOptions.RequireAccessibilityModifiers);
+
             foreach (var rule in rules)
             {
                 if (rule.SymbolSpecification.AppliesTo(SymbolKind.Field, Accessibility.Private))
                 {
                     var uniqueName = GenerateUniqueName(parameter, parameterNameParts, rule);
 
+                    var accessibilityLevel = Accessibility.Private;
+                    if (requireAccessibilityModifiers.Value == AccessibilityModifiersRequired.Never || requireAccessibilityModifiers.Value == AccessibilityModifiersRequired.OmitIfDefault)
+                    {
+                        // C#: always remove modifier for fields if "omit if default" or "never" option is set
+                        if (functionDeclaration.Language == LanguageNames.CSharp)
+                        {
+                            accessibilityLevel = Accessibility.NotApplicable;
+                        }
+                        // VB: only remove modifier for fields if "omit if default" or "never" option is set and code is within a class or module
+                        else if (functionDeclaration.Language == LanguageNames.VisualBasic)
```

Comment:
so, at core layers, we do not fork behavior based on language.  The correct way to do this is to introduce an abstract method to control this and then supply implementations in the C# and VB subclasses.  Thanks!